osu.Game/Screens/OnlinePlay/Multiplayer/GameplayChatDisplay.cs

```diff
@@ -37,16 +38,21 @@ public GameplayChatDisplay(Room room)
             : base(room, leaveChannelOnDispose: false)
         {
             RelativeSizeAxes = Axes.X;
-
             Background.Alpha = 0.2f;
+        }
 
-            TextBox.PlaceholderText = ChatStrings.InGameInputPlaceholder;
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
+        {
+            resetPlaceholderText();
             TextBox.Focus = () => TextBox.PlaceholderText = Resources.Localisation.Web.ChatStrings.InputPlaceholder;
             TextBox.FocusLost = () =>
             {
-                TextBox.PlaceholderText = ChatStrings.InGameInputPlaceholder;
+                resetPlaceholderText();
                 expandedFromTextBoxFocus.Value = false;
             };
+
+            void resetPlaceholderText() => TextBox.PlaceholderText = ChatStrings.InGameInputPlaceholder(config.LookupKeyBindings(GlobalAction.ToggleChatFocus));
```

Comment:
my question here would be why is this going through `OsuConfigManager` in the first place. `RealmKeyBindingStore.GetReadableKeyCombinationsFor(GlobalAction)` already exists, is available through DI, and `OsuConfigManager` essentially proxies to it, with a small asterisk of having a fallback for no combination bound (in which case it's arguable that better local handling could be done here, by not having the placeholder at all).

Reply:
I can try updating all usages in this PR. There were a handful of other places so I was hesitant to expand scope further.

Reply:
I don't follow. isn't `OsuConfigManager.LookupKeyBindings()` a new method added in this PR with only one usage - namely this specific one?

Reply:
It might look that way because I used the same method name, but:

https://github.com/ppy/osu/blob/345fb60679c3017f75b1b2d9dd54c210fde50a25/osu.Game/Overlays/OSD/SpeedChangeToast.cs#L13

https://github.com/ppy/osu/blob/de78518fea14b4c12c5a2db4bc74d65335f05521/osu.Game/Rulesets/Edit/ComposerDistanceSnapProvider.cs#L317

https://github.com/ppy/osu/blob/4b038c37627bcabce7b299f8d60995dbb455be40/osu.Game/Screens/OnlinePlay/Multiplayer/GameplayChatDisplay.cs#L55

https://github.com/ppy/osu/blob/252084de245263c580130d579d66759c03dbf351/osu.Game/Screens/Play/HUDOverlay.cs#L217

😅

Reply:
oh so that was all using the jammed-in func. dear lord x2.

I'd be in favour of moving all of that to use the keybinding store for sure. not sure that's enough to kill all of the weird func passing in `OsuGameBase` but it may at least kill one.